Mathlib/LinearAlgebra/Finsupp/VectorSpace.lean

```diff
@@ -49,6 +49,12 @@ theorem linearIndependent_single {φ : ι → Type*} {f : ∀ ι, φ ι → M}
       rw [span_le, range_coe]
       apply range_comp_subset_range _ (lsingle i)
 
```

Comment:
Is
```lean
theorem linearIndependent_single_iff {φ : ι → Type*} {f : ∀ ι, φ ι → M} :
    (LinearIndependent R fun ix : Σi, φ i => single ix.1 (f ix.1 ix.2)) ↔
      ∀ i, LinearIndependent R (f i) := by
  refine ⟨fun h i => ?_, linearIndependent_single⟩
  replace h := h.comp _ (sigma_mk_injective  (i := i))
  exact .of_comp (Finsupp.lsingle i) h
```
helpful towards generalizing here?

Reply:
What's the generalisation you have in mind?